### PR TITLE
Vlad rossohin/user profile

### PR DESCRIPTION
**Pull request by VladRossohin**
Please view my pull request! 
I completed add services in BL

### CODE REVIEW

File: DIMS/DIMS.BL/Interfaces/IDirectionService.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public interface IDirectionService
+    {
+        void SaveDirection(DirectionDTO direction);
+        DirectionDTO GetDirection(int? id);
+        void UpdateDireciton(DirectionDTO direction);
+        void DeleteDirection(int? id);
+
+        IEnumerable<DirectionDTO> GetDirections();
+        void Dispose();
```
Comment (maximsan) [inline DIMS/DIMS.BL/Interfaces/IDirectionService.cs]: I want to propose you to extract all common methods to one common generic IService<T> interface (SaveItem, GetItem, UpdateItem, DeleteItem, GetItems) and reduce our repetitive code.

File: DIMS/DIMS.BL/Interfaces/IvUserTrackService.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace HIMS.BL.Interfaces
+{
+    public interface IvUserTrackService
+    {
+        vUserTrackDTO GetVUserTrack(int? id);
+        ICollection<vUserTrackDTO> GetVUserTracks();
+
+        void Dispose();
```
Comment (maximsan) [inline DIMS/DIMS.BL/Interfaces/IvUserTrackService.cs]: I guess it's valid to do the same for views services and create IvService<T>

File: DIMS/DIMS.BL/Services/TaskTrackService.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+            var task = Database.TaskTracks.Get(id.Value);
+
+            if (task == null)
+                throw new ValidationException($"The task track with id = {id.Value} was not found", String.Empty);
+
+            return Mapper.Map<TaskTrack, TaskTrackDTO>(task);
+        }
+
+        public ICollection<TaskTrackDTO> GetTaskTracks()
```
Comment (maximsan) [inline DIMS/DIMS.BL/Services/TaskTrackService.cs]: IEnumerable

File: DIMS/DIMS.BL/Services/vUserTaskService.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+            var _vUserTask = Database.vUserTasks.Get(id.Value);
+
+            if (_vUserTask == null)
+                throw new ValidationException($"The view user task with id = {id.Value} was not found", String.Empty);
+
+            return Mapper.Map<vUserTask, vUserTaskDTO>(_vUserTask);
+        }
+
+        public ICollection<vUserTaskDTO> GetVUserTasks()
```
Comment (maximsan) [inline DIMS/DIMS.BL/Services/vUserTaskService.cs]: IEnumerable

File: DIMS/DIMS.BL/Services/vUserTrackService.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+            var _vUserTrack = Database.vUserTracks.Get(id.Value);
+
+            if (_vUserTrack == null)
+                throw new ValidationException($"The view user track with id = {id.Value} was not found", String.Empty);
+
+            return Mapper.Map<vUserTrack, vUserTrackDTO>(_vUserTrack);
+        }
+
+        public ICollection<vUserTrackDTO> GetVUserTracks()
```
Comment (maximsan) [inline DIMS/DIMS.BL/Services/vUserTrackService.cs]: IEnumerable

File: DIMS/DIMS.Tests/DAL.Data/Repositories/StubSampleRepository.cs [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+using System.Collections.Generic;
+using HIMS.EF.DAL.Data;
+using System.Data.Entity;
+using System.Linq;
+using System.Data.Entity.Core.EntityClient;
+
+namespace HIMS.Tests
+{
+    public class StubSampleRepository : IRepository<Sample>
+    {
```
Comment (maximsan) [inline DIMS/DIMS.Tests/DAL.Data/Repositories/StubSampleRepository.cs]: please, remove it if you do not use it

File: DIMS/DIMS.Tests/DAL.BL/Services/StubUserProfileService.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HIMS.Tests.DAL.BL.Services
+{
+    public class StubUserProfileService : IUserProfileService
+    {
```
Comment (maximsan) [inline DIMS/DIMS.Tests/DAL.BL/Services/StubUserProfileService.cs]: Do we need it?

**Comment by maximsan** [comment]
Please, clean code. I will merge it, but we need to improve our unit tests. They look terrible)

File: DIMS/DIMS.BL/DTO/vTaskDTO.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace HIMS.BL.DTO
+{
+    public class vTaskDTO
+    {
+        public int TaskId { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public System.DateTime StartDate { get; set; }
+        public System.DateTime DeadlineDate { get; set; }
```
[the remark this answers is not in the document]
Comment (VladRossohin) [reply DIMS/DIMS.BL/DTO/vTaskDTO.cs]: done

File: DIMS/DIMS.BL/Interfaces/IvUserProfileService.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HIMS.BL.Interfaces
+{
+    public interface IvUserProfileService
+    {
+        vUserProfileDTO GetVUserProfile(int? id);
+        ICollection<vUserProfileDTO> GetVUserProfiles();
```
[the remark this answers is not in the document]
Comment (VladRossohin) [reply DIMS/DIMS.BL/Interfaces/IvUserProfileService.cs]: ok, i changed it in all services

File: DIMS/DIMS.BL/Interfaces/IDirectionService.cs (hunk 1 above)
Comment (VladRossohin) [reply DIMS/DIMS.BL/Interfaces/IDirectionService.cs]: done

File: DIMS/DIMS.BL/Interfaces/IvUserTrackService.cs (hunk 2 above)
Comment (VladRossohin) [reply DIMS/DIMS.BL/Interfaces/IvUserTrackService.cs]: done

**Pull request merged**